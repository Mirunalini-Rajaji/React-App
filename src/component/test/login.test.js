import React from 'react';
import ReactDOM from 'react-dom'
import Login from '../login/login'
import LoginTest from '../login/logintest'
import { Link } from "react-router-dom";
import { mount, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {MemoryRouter as Router} from 'react-router-dom'
import {  render } from '@testing-library/react';
import renderer from "react-test-renderer";
import '@testing-library/jest-dom/extend-expect'
configure({ adapter: new Adapter() })

describe("login",()=>{
    
it('login renders without crashing', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<Router><Login></Login></Router>, div)
})



it(" check email", () => {
    const wrapper = mount(<input type="text" placeholder="Email Address" />);
    const input = wrapper.find("input");
    expect(input.prop("type")).toEqual("text");
    expect(input.prop("placeholder")).toEqual("Email Address");
  });
it(" check password", () => {
    const wrapper = mount(<input type="password" placeholder="Password" />);
    const input = wrapper.find("input");
    expect(input.prop("type")).toEqual("password");
    expect(input.prop("placeholder")).toEqual("Password");
});
it('check if account text renders correctly', ()=>{
    const {getByTestId} = render(<Router><Login></Login></Router>)
    expect(getByTestId('p')).toHaveTextContent('Dont have an account?')
})
it('check link',()=>{
    const link = renderer.create(<Router><Link to="/createaccount"></Link></Router>).toJSON()
    expect(link).toMatchSnapshot()
})
it("Api Testing for matching password", async function () {
    const response = new LoginTest();
    console.warn(await response.api());
    const data = await response.api();
    expect(data[0].pwd).toEqual("adc");
   })

})

