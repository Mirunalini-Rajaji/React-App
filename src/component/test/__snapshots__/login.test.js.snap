// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check link 1`] = `
<a
  href="/createaccount"
  onClick={[Function]}
/>
`;

exports[`login check link 1`] = `
<a
  href="/createaccount"
  onClick={[Function]}
/>
`;
